src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1269,10 +1269,36 @@ public int Send(ReadOnlySpan<byte> buffer, SocketFlags socketFlags, out SocketEr
 
         public void SendFile(string? fileName)
         {
-            SendFile(fileName, null, null, TransmitFileOptions.UseDefaultWorkerThread);
+            SendFile(fileName, ReadOnlySpan<byte>.Empty, ReadOnlySpan<byte>.Empty, TransmitFileOptions.UseDefaultWorkerThread);
         }
 
         public void SendFile(string? fileName, byte[]? preBuffer, byte[]? postBuffer, TransmitFileOptions flags)
+        {
+            SendFile(fileName, preBuffer.AsSpan(), postBuffer.AsSpan(), flags);
```

Comment:
Does this handle null values for pre/postBuffer correctly?

Do we have tests for that?

Reply:
A `null` array will become an empty span, which is checked in the control flow.

No explicit test, but it's covered by https://github.com/dotnet/runtime/blob/5a4483eb1c0918542298d1864b7233f962952880/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendFile.cs#L137-L147